providers/k8s/resources/discovery.go

```diff
@@ -1157,3 +1178,17 @@ func assetName(ns, name string) string {
 	}
 	return ns + "/" + name
 }
+
+func ownerReferencesFilter(refs []metav1.OwnerReference, filter map[string]struct{}) bool {
```

Comment:
can we change the filter to a `[]string` instead? i see no reason why it should be a map. you can even do a `string...` param to allow for something like `ownerReferenceFilters(obj,"DaemonSet","CronJob")`

Reply:
you can then either build the map internally or just use the `stringx.Contains` func which can check if the `ref.Kind` is in that list

Reply:
Map is just golang's idiomatic way to implement Set, which does save unnecessary iterations when searching for a string inside a slice.

Reply:
@imilchev @preslavgerchev no problem, done